client/src/components/SignupTwo.js

```diff
@@ -2,190 +2,233 @@
-					align="center"
-					className={classes.modalTitle}
-				>
-					Sign Up
-				</Typography>
-				<Typography
-					component="p"
-					variant="subtitle1"
-					align="center"
-					className={classes.modalSubtitle}
-				>
-					Please select your favorite travel destinations
-				</Typography>
-				<Grid container spacing={1}>
-					{travelList.map((place) => (
-						<Grid key={place} item xs={12}>
-							<InputBase
-								name={place}
-								id={place}
-								value={place}
-								readOnly
-								fullWidth
-								classes={{ root: classes.inputBase, input: classes.input }}
-								startAdornment={
-									<InputAdornment position="start">
-										<LocationOnOutlinedIcon
-											classes={{ root: classes.locationIcon }}
-										/>
-									</InputAdornment>
-								}
-								endAdornment={
-									<InputAdornment position="end">
-										<IconButton
-											aria-label={`remove ${place}`}
-											onClick={() => handleDelete(place)}
-											size="small"
-										>
-											<CloseIcon classes={{ root: classes.closeIcon }} />
-										</IconButton>
-									</InputAdornment>
-								}
-							/>
-						</Grid>
-					))}
-				</Grid>
-				<Box textAlign="center" mt={3} mb={2}>
-					{open ? (
-						<Input
-							id="add-destination"
-							color="secondary"
-							onChange={handleTextChange}
-							value={destination}
-							endAdornment={
-								<InputAdornment position="end">
-									<IconButton
-										aria-label="add destination"
-										onClick={handleAdd}
-										edge="end"
-									>
-										<AddIcon />
-									</IconButton>
-								</InputAdornment>
-							}
-						/>
-					) : (
-						<Button className={classes.link} onClick={openAdd}>
-							Add more
-						</Button>
-					)}
-				</Box>
-				<Box textAlign="center">
-					<Button
-						type="submit"
-						variant="contained"
-						color="primary"
-						size="large"
-						classes={{ root: classes.modalSubmit }}
-						onClick={handleSubmit}
-					>
-						Continue
-					</Button>
-				</Box>
-			</div>
-			{/* Need to change the footer if user is automatically signed in. */}
-			<div className="modal-footer">
-				<Typography
-					component="p"
-					variant="subtitle1"
-					align="center"
-					className={classes.modalFooter}
-				>
-					Already have an account?{" "}
-					<Link to="/signin" className={classes.link}>
-						Sign In
-					</Link>
-				</Typography>
-			</div>
-		</Container>
-	);
+  const handleSubmit = async (event) => {
+    const userId = user?._id;
+    event.preventDefault();
+    try {
+      const res = await fetch(`/api/users/${userId}/favorite-cities`, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({ cities: travelList }),
+      });
+      const resData = await res.json();
+      if (resData.message === "Updated") {
+        window.location.href = "/explore";
+      } else {
```

Comment:
Have you tried redirecting to the Explore page with React Router's History object?

Reply:
The page needs to be reloaded, so the context is updated with the new user object.

Reply:
Why does this page need to be reloaded?

Reply:
So the context gets updated.